### ISSUE TITLE
[bitnami/mediawiki] 404 errors when overriding MEDIAWIKI_WIKI_PREFIX

### ISSUE THREAD

**Issue by dunn** [bitnami/containers#64097]
### Name and Version

bitnami/mediawiki:1

### What architecture are you using?

amd64

### What steps will reproduce the bug?

set a non-default value for the prefix in the docker compose `environment`:
```
- MEDIAWIKI_WIKI_PREFIX=/something
```
then run `docker compose up`

### What is the expected behavior?

MediaWiki is accessible at `/something` instead of `/wiki`

### What do you see instead?

Every page under `/something` returns a 404 error:
```
mediawiki-1  | mediawiki 22:25:41.93 INFO  ==> Running MediaWiki install script
mediawiki-1  | mediawiki 22:25:43.06 INFO  ==> Configuring MediaWiki settings
mediawiki-1  | mediawiki 22:25:43.07 INFO  ==> Setting MediaWiki short URLs
mediawiki-1  | mediawiki 22:25:43.08 INFO  ==> Persisting MediaWiki installation
mediawiki-1  | mediawiki 22:25:44.95 INFO  ==> ** MediaWiki setup finished! **
mediawiki-1  |
mediawiki-1  | mediawiki 22:25:44.95 INFO  ==> ** Starting Apache **
mediawiki-1  | [Sat Mar 23 22:25:45.180354 2024] [mpm_prefork:notice] [pid 1] AH00163: Apache/2.4.58 (Unix) OpenSSL/3.0.11 configured -- resuming normal operations
mediawiki-1  | [Sat Mar 23 22:25:45.180392 2024] [core:notice] [pid 1] AH00094: Command line: '/opt/bitnami/apache/bin/httpd -f /opt/bitnami/apache/conf/httpd.conf -D FOREGROUND'
mediawiki-1  | 192.168.160.1 - - [23/Mar/2024:22:26:06 +0000] "GET / HTTP/1.1" 301 -
mediawiki-1  | 192.168.160.1 - - [23/Mar/2024:22:26:07 +0000] "GET /something/Main_Page HTTP/1.1" 404 196
mediawiki-1  | 192.168.160.1 - - [23/Mar/2024:22:26:07 +0000] "GET /favicon.ico HTTP/1.1" 404 196
```

**Comment by dunn** [comment]
afaict the `RewriteRule` need to match the prefix, instead of being hardcoded to `/wiki`: https://github.com/bitnami/containers/blob/main/bitnami/mediawiki/1/debian-12/rootfs/opt/bitnami/scripts/mediawiki/postunpack.sh#L50

But since `postunpack.sh` is run during the image build, it can't be overridden with environment variables.

**Comment by carrodher** [comment]
Thank you for opening this issue and submitting the associated Pull Request. Our team will review and provide feedback. Once the PR is merged, the issue will automatically close.

Your contribution is greatly appreciated!

**Comment by dunn** [comment]
un-stale

**Comment by dunn** [comment]
\>:(

**Comment by Ducatel** [comment]
Hi seems I face a very similar issue.
I tried to run mediawiki 1.41.1 under kubernetes with env variables  `MEDIAWIKI_HOST`, `MEDIAWIKI_WIKI_PREFIX` and I'm redirected without any domain when I reach the `/`

![image](https://github.com/bitnami/containers/assets/1184332/b202cc80-2b38-4be4-afee-ecb4919c0c24)

**Comment by migruiz4** [comment]
Hi @Ducatel,

The katest version of the `bitnami/mediawiki` image now includes the fix (`1.41.1-debian-12-r8`).

**Comment by Ducatel** [comment]
Ok So still not working for me. let's open a dedicated one

**Comment by jotamartos** [comment]
Hi

As I just mentioned in the PR, these changes are breaking the functionality of the container, you get several redirect loops when you try to access the app or work with it. This is preventing our test to pass and update the Chart solution.

We are planing some changes to make the solution work again and they will be included in the next release:

- Recover the old logic to configure the $wgServer parameter in the conf file
- The $wgCanonicalServer parameter uses that configuration by default according to the official documentation: https://www.mediawiki.org/wiki/Manual:$wgCanonicalServer
- RewriteRules will be updated to use the ones mentioned in the official documentation https://www.mediawiki.org/wiki/Manual:Short_URL/wiki/Page_Title_--_.htaccess

If you continue running into issues when running the app behind a proxy, please do create another PR to update the configuration taking these changes into account.

**Comment by jotamartos** [comment]
Changes were applied here: https://github.com/bitnami/containers/pull/67621
